Several `to_pandas()` tests are failing for the most recent pandas

Several pandas tests for keyed frames are failing for pandas `1.3.5` and python `3.9`
```
tests/frame/test-to-pandas.py::test_topandas_keyed FAILED
tests/frame/test-to-pandas.py::test_topandas_keyed2 FAILED
```

The error is 
```
    @pandas_test
    def test_topandas_keyed(pd):
        DT = dt.Frame(A=[3, 5, 9, 11], B=[7, 14, 1, 0], R=['va', 'dfkjv', 'q', '...'])
        DT.key = 'A'
>       pf = DT.to_pandas()

tests/frame/test-to-pandas.py:118: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../py39/lib/python3.9/site-packages/pandas/core/indexes/base.py:469: in __new__
    arr = klass._ensure_array(arr, dtype, copy)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

cls = <class 'pandas.core.indexes.numeric.Int64Index'>, data = array([ 3,  5,  9, 11], dtype=int32), dtype = dtype('int64'), copy = None

    @classmethod
    def _ensure_array(cls, data, dtype, copy: bool):
        """
        Ensure we have a valid array to pass to _simple_new.
        """
        cls._validate_dtype(dtype)
    
        if not isinstance(data, (np.ndarray, Index)):
            # Coerce to ndarray if not already ndarray or Index
            if is_scalar(data):
                raise cls._scalar_data_error(data)
    
            # other iterable of some kind
            if not isinstance(data, (ABCSeries, list, tuple)):
                data = list(data)
    
            data = np.asarray(data, dtype=dtype)
    
        if issubclass(data.dtype.type, str):
            cls._string_data_error(data)
    
        dtype = cls._ensure_dtype(dtype)
    
        if copy or not is_dtype_equal(data.dtype, dtype):
>           subarr = np.array(data, dtype=dtype, copy=copy)
E           ValueError: NoneType copy mode not allowed.

../../py39/lib/python3.9/site-packages/pandas/core/indexes/numeric.py:164: ValueError
```

--- a/src/core/frame/to_pandas.cc
+++ b/src/core/frame/to_pandas.cc
@@ -43,7 +43,7 @@ oobj Frame::to_pandas(const XArgs&) {
     olist indices(nkeys);
     for (size_t i = 0; i < nkeys; i++) {
       auto column = robj(this).invoke("to_numpy", {None(), oint(i)});
-      auto index_item = pd_Index.call({column, None(), None(), names[i]});
+      auto index_item = pd_Index.call({column, None(), False(), names[i]});
       indices.set(i, std::move(index_item));
     }
     if (nkeys == 1) {

